### ISSUE TITLE
[mgtv] An extractor error has occurred (update on version 2020.11.12).

### ISSUE THREAD

**Issue by TurnOffNOD** [ytdl-org/youtube-dl#26415]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version 2020.07.28
Update: youtube-dl version 2020.09.14, problem still exists.
Update: youtube-dl version **2020.11.12**, problem still exists
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log (version 2020.11.12)
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--no-geo-bypass', '--verbose', 'https://www.mgtv.com/b/334727/10259082.html']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.11.12
[debug] Python version 3.8.6 (CPython) - Linux-4.19.128-microsoft-standard-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.3.1-5, ffprobe 4.3.1-5, rtmpdump 2.4
[debug] Proxy map: {}
[MGTV] 10259082: Downloading JSON metadata
[MGTV] 10259082: Downloading JSON metadata
ERROR: An extractor error has occurred. (caused by KeyError('stream_domain')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 532, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/mgtv.py", line 61, in _real_extract
    stream_domain = stream_data['stream_domain'][0]
KeyError: 'stream_domain'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 532, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/mgtv.py", line 61, in _real_extract
    stream_domain = stream_data['stream_domain'][0]
KeyError: 'stream_domain'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 545, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
youtube_dl.utils.ExtractorError: An extractor error has occurred. (caused by KeyError('stream_domain')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

##### Verbose log (version 2020.09.14)
```######
$ youtube-dl --no-geo-bypass --verbose https://www.mgtv.com/b/334727/9996419.html
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--no-geo-bypass', '--verbose', 'https://www.mgtv.com/b/334727/9996419.html']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.09.14
[debug] Python version 3.8.6 (CPython) - Linux-4.19.128-microsoft-standard-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.3.1-4, ffprobe 4.3.1-4, rtmpdump 2.4
[debug] Proxy map: {}
[MGTV] 9996419: Downloading JSON metadata
[MGTV] 9996419: Downloading JSON metadata
ERROR: An extractor error has occurred. (caused by KeyError('data')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/mgtv.py", line 56, in _real_extract
    stream_data = self._download_json(
KeyError: 'data'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/mgtv.py", line 56, in _real_extract
    stream_data = self._download_json(
KeyError: 'data'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 543, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
youtube_dl.utils.ExtractorError: An extractor error has occurred. (caused by KeyError('data')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

##### Verbose log (version 2020.07.28)
```######
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '--no-geo-bypass', 'https://www.mgtv.com/b/334727/9680012.html']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2020.07.28
[debug] Python version 3.8.5 (CPython) - Linux-4.19.104-microsoft-standard-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.3.1-2, ffprobe 4.3.1-2, rtmpdump 2.4
[debug] Proxy map: {}
[MGTV] 9680012: Downloading JSON metadata
[MGTV] 9680012: Downloading JSON metadata
ERROR: An extractor error has occurred. (caused by KeyError('data')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl ith the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/mgtv.py", line 56, in _real_extract
    stream_data = self._download_json(
KeyError:** 'data'
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/mgtv.py", line 56, in _real_extract
    stream_data = self._download_json(
KeyError: 'data'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 543, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
youtube_dl.utils.ExtractorError: An extractor error has occurred. (caused by KeyError('data')); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

## Description

Hi, mgtv (mgtv.com) is a website in China mainland, and I am also in China mainland, so I have to add "--no-geo-bypass" parameter, or it will end up in download error. 

I guess it will use --geo-bypass on this website by default? This result in a IP outside China mainland and cause problems.

However, this "An extractor error has occurred." has nothing to do with the parameter, because there was no problem last week (accurately on Aug. 16, UTC+8 timezone).

**Comment by 20NE** [comment]
Same issue inside China!

**Comment by tpennycook** [comment]
Same issue in Europe with mgtv. Mgtv was working with youtube-dl version 2020.07.28 until quite recently (with the --no-geo-bypass option, and at least sometimes needing the --cookies option for VIP content). 

In my case using:
[debug] youtube-dl version 2020.07.28
[debug] Python version 3.8.5 (CPython) - macOS-10.15.6-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3, ffprobe 4.3, rtmpdump 2.4

**Comment by TurnOffNOD** [comment]
Hi, youtube-dl version 2020.09.14, problem still exists. The main issue is updated.

**Comment by 20NE** [comment]
Still unfixed. youtube-dl version 2020.11.01.1

**Comment by TurnOffNOD** [comment]
> 
> 
> Still unfixed. youtube-dl version 2020.11.01.1

Still unfixed in youtube-dl version 2020.11.12

**Comment by lirc572** [comment]
Same here. It seems that mgtv extractor needs to be upgraded.

**Comment by TurnOffNOD** [comment]
Hi, problem solved in version 2020.11.21.1, thanks.
